### PR TITLE
Fix/exclude undefined detections

### PR DESCRIPTION
**Pull request by micaelmbagira**
Before we integrate the binary data in our snapshot construction, we used to add a person in the snapshot [only if he/she has an age and a gender](https://github.com/advertima/io/blob/0.3.x/src/poi/POISnapshot.ts#L96-L98).

This PR adds this check to our new implementation. We had some scenarios where the binary data were not stable and even if the person had an age and gender in previous detections, new detections with `age: undefined` would update the person.

### CODE REVIEW

File: src/poi/POISnapshot.ts [hunk 1]
```diff
@@ -205,8 +205,13 @@ export class POISnapshot {
     if (obj.json !== undefined && obj.binary !== undefined) {
       const person = PersonDetection.fromMessage(obj.json, obj.binary);
 
-      this.persons.set(obj.json.personId, person);
-      this.personsByTtid.set(obj.binary.personAttributes.ttid, person);
+      if (
```
[the remark this answers is not in the document]
Comment (fbobbio) [reply src/poi/POISnapshot.ts]: Sounds good if we have no impact in the rest of the infrastructure.

@nicojedlicka can you refresh our minds about how will the Analytics FW respond to POI snapshots flowing without age and/or gender?

[the remark this answers is not in the document]
Comment (nicojedlicka) [reply src/poi/POISnapshot.ts]: @fbobbio ATM we will discard records that don't contain AGE / GENDER, so we will need to update that as well.

[the remark this answers is not in the document]
Comment (fbobbio) [reply src/poi/POISnapshot.ts]: Ok then we are starting to have spread around, lets take it into consideration and soon discuss how to approach depending on the reqs that come from bus guys

[the remark this answers is not in the document]
Comment (micaelmbagira) [reply src/poi/POISnapshot.ts]: I am not sure what's the conclusion, should we keep the change here or have it in every service (right now Analytics and Decision)?

[the remark this answers is not in the document]
Comment (fbobbio) [reply src/poi/POISnapshot.ts]: @micaelmbagira you are right.

Lets keep it here, we have a task to change this in next sprint.

**Pull request merged**